### ISSUE TITLE
Integrate setting the prompt so that the user sees they are in a pixi shell

### ISSUE THREAD

**Issue by ruben-arts** [prefix-dev/pixi#219]
When moving into a `pixi shell` it can be convenient to see that the shell is active. 
```shell
[rarts@fedora test] $ pixi shell
(test) [rarts@fedora test] $ exit
[rarts@fedora test] $ 
```

**NOTE:** to get this to work you have to run `pixi completion` in your shells activation script, e.g. `.bashrc` or `.zshrc`. There are examples in the [docs](https://beta.prefix.dev/docs/pixi/overview#autocompletion)

Currently supported:
|Shell|Supported|
|-------|----------------|
| Bash | [:white_check_mark:](https://github.com/prefix-dev/pixi/pull/385) |
| Zsh | [:white_check_mark:](https://github.com/prefix-dev/pixi/pull/385) |
| Fish | [:white_check_mark:](https://github.com/prefix-dev/pixi/pull/385) |
| Xonsh | [:white_check_mark:](https://github.com/prefix-dev/pixi/pull/385) |
| Powershell | [:white_check_mark:](https://github.com/prefix-dev/pixi/pull/385)  |
| CMD.exe | [:white_check_mark:](https://github.com/prefix-dev/pixi/pull/385) |

**Comment by kolibril13** [comment]
Yep, I totally agree with this, I think this is super useful! For me, this does not work in Zsh on an Mac M1, as can be seen in this screenshot.


<img width="1122" alt="image" src="https://github.com/prefix-dev/pixi/assets/44469195/9140ec13-28df-4c62-bd0b-40d1688d2585">

Also, I'm wondering why it's possible to activate the shell with micromaba and conda, as can also be seen in the screenshot.
With that approach, I however get the full path `(/Users/jan-hendrik/projects/trying-pixi/.pixi/env)` , where it would be nice to only get the relative path `(.pixi/env)`

**Comment by traversaro** [comment]
> Currently supported:

Just to understand, the feature is expected to work on bash/zsh with pixi 0.2.0 ? I am asking as it is not working for me on Bash/WSL2, so if it useful I can provide debug info.

**Comment by ruben-arts** [comment]
Just to be sure I added a note in this issue, that you need to add the `pixi completion` to your rc files. Are you @traversaro @kolibril13 both running that resulting script before you run `pixi shell` ?

This is my result using `v0.2.0`:
```bash
[rarts@fedora test]$ pixi shell
(test) [rarts@fedora test]$ exit 
exit
[rarts@fedora test]$ 
```

@kolibril13 The fact that all tools can activate the environment is because we all implement the "conda spec" so its "just" a conda environment. pixi does nothing special to those environments. The magic is in the interaction with those environments.

**Comment by ruben-arts** [comment]
This has been fixed in #385
